aopy/utils/base.py

```diff
@@ -765,4 +765,34 @@ def convert_port_number(port_number, datatype='ap'):
     letter = chr(ord('@')+port_number)
     probe_dir = f'Neuropix-PXI-100.Probe{letter}-{datatype.upper()}'
     
-    return probe_dir
\ No newline at end of file
+    return probe_dir
+
+def multiply_mat_batch(data, mat, save_path, scale = 1, max_memory_gb = 1., dtype='int16'):
+    '''
+    Multiply a matrix to data in each batch to save memory. The result is saved in save_path.
+    This function can be used to multiply an inverse matrix by spike band time series.
+    
+    Args:
+        data (nt, nch): neural data. This should be a memory mapping array.
+        mat (anysize, nch): matrix to multiply by data
+        save_path (str): file path to save destriped lfp data
+        scale (float, optional): Scaling factor to multiply by data. 1/200 is necessary for whitened data in kilosort4. default is 1.
+        max_memory_gb (float): memory size in GB to determine batch size. default is 1.0 GB.
+        dtype (str, optional): dtype for data. default is int16.
+    
+    Returns:
+        None
+    
+    '''
+        
+    n_samples, n_channels = data.shape
+    batch_size = int (max_memory_gb*1e9 / (n_channels*np.dtype(dtype).itemsize))
+    Nbatches = np.ceil(n_samples/batch_size).astype(int)
+
+    # Multiply matrix in each batch to save memory
+    x = np.memmap(save_path, dtype=dtype, mode='w+', shape=(n_samples, n_channels))
+
```

Comment:
can you add the min batch size argument like you did in the other pull request?

Reply:
min batch size is not necessary for this function because this is just multiplication between 2 matrices.